api/v1beta1/azuremanagedcontrolplane_types.go

```diff
@@ -220,6 +232,89 @@ type AzureManagedControlPlaneSpec struct {
 	// DisableLocalAccounts disables getting static credentials for this cluster when set. Expected to only be used for AAD clusters.
 	// +optional
 	DisableLocalAccounts *bool `json:"disableLocalAccounts,omitempty"`
+
+	// SecurityProfile defines the security profile for the container service cluster.
+	// +optional
+	SecurityProfile *ManagedClusterSecurityProfile `json:"securityProfile,omitempty"`
+}
+
+// ManagedClusterSecurityProfile defines the security profile for the container service cluster.
+type ManagedClusterSecurityProfile struct {
+	// AzureKeyVaultKms defines Azure Key Vault key management service settings for the security profile.
+	// +optional
+	AzureKeyVaultKms *AzureKeyVaultKms `json:"azureKeyVaultKms,omitempty"`
+
+	// Defender defines Microsoft Defender settings for the security profile.
+	// +optional
+	Defender *ManagedClusterSecurityProfileDefender `json:"defender,omitempty"`
+
+	// ImageCleaner settings for the security profile.
+	// +optional
+	ImageCleaner *ManagedClusterSecurityProfileImageCleaner `json:"imageCleaner,omitempty"`
+
+	// Workloadidentity enables Kubernetes applications to access Azure cloud resources securely with Azure AD. Ensure to enable OIDC issuer while enabling Workload Identity
+	// +optional
+	WorkloadIdentity *ManagedClusterSecurityProfileWorkloadIdentity `json:"workloadIdentity,omitempty"`
+}
+
+// ManagedClusterSecurityProfileDefender defines Microsoft Defender settings for the security profile.
+type ManagedClusterSecurityProfileDefender struct {
+	// LogAnalyticsWorkspaceResourceID is the ID of the Log Analytics workspace that has to be associated with Microsoft Defender.
+	// When Microsoft Defender is enabled, this field is required and must be a valid workspace resource ID.
+	// +kubebuilder:validation:Required
+	LogAnalyticsWorkspaceResourceID string `json:"logAnalyticsWorkspaceResourceID"`
+
+	// SecurityMonitoring profile defines the Microsoft Defender threat detection for Cloud settings for the security profile.
+	// +kubebuilder:validation:Required
+	SecurityMonitoring ManagedClusterSecurityProfileDefenderSecurityMonitoring `json:"securityMonitoring"`
+}
+
+// ManagedClusterSecurityProfileDefenderSecurityMonitoring settings for the security profile threat detection.
+type ManagedClusterSecurityProfileDefenderSecurityMonitoring struct {
+	// Enabled enables Defender threat detection
+	// +kubebuilder:validation:Required
+	Enabled bool `json:"enabled"`
+}
```

Comment:
Same comment as https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/4245/files#r1395268980
Why are we introducing the struct and not using the boolean directly for `SecurityMonitoring`

Reply:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/4245#discussion_r1412865455